### PR TITLE
master-errata enable option

### PR DESCRIPTION
**Pull request by williamcroberts**

### CODE REVIEW

**Comment by williamcroberts** [comment]
Im going to put  f400ceb on a separate PR to 3.X and master, as we need it for bug fix.

File: lib/tpm2_errata.c [hunk 1]
```diff
@@ -143,6 +143,9 @@ void tpm2_errata_fixup(tpm2_errata_index_t index, ...) {
         return;
     }
 
+    LOG_WARN("Applying errata: \"%s\", use --disable-errata to disable!",
+            errata->name);
```
Comment (martinezjavier) [inline lib/tpm2_errata.c]: If we flip the logic to instead have an option to enable erratas, I think we should change this to LOG_INFO().

File: tools/tpm2_tool.c [hunk 2]
```diff
@@ -139,7 +139,11 @@ int main(int argc, char *argv[], char *envp[]) {
     /* TODO SAPI INIT */
     TSS2_SYS_CONTEXT *sapi_context = sapi_ctx_init(tcti);
 
-    tpm2_errata_init(sapi_context);
+    if (!flags.disable_errata) {
+        tpm2_errata_init(sapi_context);
+    } else {
+       LOG_INFO("Errata disabled");
```
Comment (martinezjavier) [inline tools/tpm2_tool.c]: And this could also be removed if we flip the logic.

File: lib/tpm2_errata.c (hunk 1 above)
Comment (williamcroberts) [reply lib/tpm2_errata.c]: Agreed.

File: tools/tpm2_tool.c (hunk 2 above)
Comment (williamcroberts) [reply tools/tpm2_tool.c]: Agreed.

**Comment by williamcroberts** [comment]
@martinezjavier I am inclined to disable by default as well, lets let this soak for a day for @jiazhang0 to respond. If no argument contrary will flip the logic tomorrow and submit on master and 3.X

File: lib/tpm2_errata.c (hunk 1 above)
Comment (williamcroberts) [reply lib/tpm2_errata.c]: There is an Errata applied statement at the end...

File: lib/tpm2_options.c [hunk 3]
```diff
@@ -252,12 +252,17 @@ tpm2_option_code tpm2_handle_options (int argc, char **argv, char **envp,
...
-        { "help",    no_argument,       NULL, 'h' },
-        { "verbose", no_argument,       NULL, 'v' },
-        { "quiet",   no_argument,       NULL, 'Q' },
-        { "version", no_argument,       NULL, 'V' },
+        { "tcti",           required_argument, NULL, 'T' },
+        { "help",           no_argument,       NULL, 'h' },
+        { "verbose",        no_argument,       NULL, 'v' },
+        { "quiet",          no_argument,       NULL, 'Q' },
+        { "version",        no_argument,       NULL, 'V' },
+        { "enable-errata", no_argument,       NULL, 'Z' },
```
Comment (williamcroberts) [inline lib/tpm2_options.c]: nit: space

**Comment by jiazhang0** [comment]
@williamcroberts @martinezjavier I totally agree with disabling errata by default because there is no way to enforce manufacturer strictly following the spec. Since we already see a broken TPM firmware, it means we will see more in future.

**Pull request merged**